l10n_it_central_journal_reportlab/wizard/print_giornale.py

```diff
@@ -0,0 +1,602 @@
+                Paragraph(_("Debit"), style_header_number),
+                Paragraph(_("Credit"), style_header_number),
+            ]
+        ]
+        return data_header
+
+    def get_initial_balance_data_report_giornale(self):
+        style_name = self.get_styles_report_giornale_line()["style_name"]
+        style_number = self.get_styles_report_giornale_line()["style_number"]
+
+        initial_balance_data = [
+            [
+                "",
+                "",
+                "",
+                "",
+                Paragraph(_("Initial Balance"), style_name),
+                Paragraph(formatLang(self.env, self.progressive_debit2), style_number),
+                Paragraph(formatLang(self.env, self.progressive_credit), style_number),
+            ]
+        ]
+        return initial_balance_data
+
+    def get_grupped_final_tables_report_giornale(
+        self, list_grupped_line, tables, start_row, width_available
+    ):
+        style_name = self.get_styles_report_giornale_line()["style_name"]
+        style_number = self.get_styles_report_giornale_line()["style_number"]
+        style_table = self.get_styles_report_giornale_line()["style_table"]
+        style_table_line_above = self.get_styles_report_giornale_line()[
+            "style_table_line_above"
+        ]
+        colwidths = self.get_colwidths_report_giornale(width_available)
+
+        previous_move_name = ""
+        list_balance = [
+            (0, 0),
+            (self.progressive_debit2, self.progressive_credit),
+        ]
+        for line in list_grupped_line:
+            start_row += 1
+            row = Paragraph(str(start_row), style_name)
+            date = Paragraph(format_date(self.env, line["date"]), style_name)
+            move = Paragraph(line["move_name"], style_name)
+            account_name = (
+                line["account_code"] + " - " + line["account_name"]
+                if line["account_code"]
+                else line["account_name"]
+            )
+            account = Paragraph(account_name, style_name)
+            name = Paragraph(line["name"], style_name)
+            # dato che nel SQL ho la somma dei crediti e debiti potrei avere
+            # che un conto ha sia debito che credito
+            lines_data = []
+            if line["debit"] > 0:
+                debit = Paragraph(formatLang(self.env, line["debit"]), style_number)
+                credit = Paragraph(formatLang(self.env, 0), style_number)
+                list_balance.append((line["debit"], 0))
+                lines_data.append([[row, date, move, account, name, debit, credit]])
+            if line["credit"] > 0:
+                debit = Paragraph(formatLang(self.env, 0), style_number)
+                credit = Paragraph(formatLang(self.env, line["credit"]), style_number)
+                list_balance.append((0, line["credit"]))
+                lines_data.append([[row, date, move, account, name, debit, credit]])
```

Comment:
c'è un motivo per cui sono stati fatti due if in cascata quando si potrebbe eseguire questo codice?

Da questo:

```
            if line["debit"] > 0:
                debit = Paragraph(formatLang(self.env, line["debit"]), style_number)
                credit = Paragraph(formatLang(self.env, 0), style_number)
                list_balance.append((line["debit"], 0))
                lines_data.append([[row, date, move, account, name, debit, credit]])
            if line["credit"] > 0:
                debit = Paragraph(formatLang(self.env, 0), style_number)
                credit = Paragraph(formatLang(self.env, line["credit"]), style_number)
                list_balance.append((0, line["credit"]))
                lines_data.append([[row, date, move, account, name, debit, credit]])
```

a questo:
```
            debit = Paragraph(formatLang(self.env, line["debit"]), style_number)
            credit = Paragraph(formatLang(self.env, line["credit"]), style_number)
            list_balance.append((line["credit"], line["credit"]))
            lines_data.append([[row, date, move, account, name, debit, credit]])
```